Cabal/Distribution/FieldGrammar/Parsec.hs

```diff
@@ -155,37 +160,37 @@ instance FieldGrammar ParsecFieldGrammar where
     uniqueFieldAla fn _pack _extract = ParsecFG (Set.singleton fn) Set.empty parser
       where
         parser v fields = case Map.lookup fn fields of
-            Nothing -> parseFatalFailure zeroPos $ show fn ++ " field missing"
-            Just [] -> parseFatalFailure zeroPos $ show fn ++ " field missing"
-            Just [x] -> parseOne v x
-            Just xs -> do
-                warnMultipleSingularFields fn xs
-                last <$> traverse (parseOne v) xs
+            Nothing     -> parseFatalFailure zeroPos $ show fn ++ " field missing"
+            Just []     -> parseFatalFailure zeroPos $ show fn ++ " field missing"
+            Just [x]    -> parseOne v x
+            Just (x:xs) -> do
```

Comment:
```
(_:x:xs) ->
```

The first one is definitely not the last one here! :) (I don't know why i didn't `last xs`

Reply:
Don't I need to keep the first one to emit the warning?  I guess I can drop the `x` in `x:|xs`.

Reply:
Ah true, then it will be

```
Just xs@(_:y:ys) -> do
    warnMultipleSingularFields fn xs
    NonEmpty.last <$> traverse (parseOne v) (y:|ys)